### PR TITLE
fix: resolves issue when using ApplicationClient.prepare with a custom TransactionSigner, resolves #213

### LINKED ISSUE #213: ApplicationClient.get_sender() does not handle custom TransactionSigner implementations properly
Below is the ApplicationClient.get_sender() code. I have a custom TransactionSigner implementation that I want to use. Even though, I specify a sender and signer when I create an ApplicationClient instance. I expect the sender that is explicitly specified to be used as the default sender. However, the current logic ignores the sender if the signer is set.

Even though I have specified a sender (and signer), I have to specify the sender for each ApplicationClient call, e.g.

If I try to create the contract with the following line of code:
`app_client.create()`
it fails with the error msg : "No sender provided"

To make it work, I have to specify the sender again, even though I specified it at construction time as the default sender:
`app_client.create(sender=sender)`

### PR DESCRIPTION
**Pull request by daniel-makerx**

### CODE REVIEW

File: pyproject.toml
Comment (daniel-makerx) [inline pyproject.toml:22]: Will do a proper release before merging this PR

File: tests/client/application_client_test.py
Comment (barnjamin) [inline tests/client/application_client_test.py:8]: We're importing from transaction below, can we just import Transaction from there as well?

Comment (daniel-makerx) [reply tests/client/application_client_test.py]: Good spot!!, done

**Review by barnjamin — APPROVED** [review]

File: pyproject.toml
Comment (barnjamin) [reply pyproject.toml:22]: I've approved but will hold off merging until this one is resolved

Comment (daniel-makerx) [reply pyproject.toml]: Updated to 1.0.2 proper

**Pull request merged**
